mypy/argmap.py

```diff
@@ -193,10 +200,13 @@ def expand_actual_type(self,
                     formal_name = (set(actual_type.items.keys()) - self.kwargs_used).pop()
                 self.kwargs_used.add(formal_name)
                 return actual_type.items[formal_name]
-            elif (isinstance(actual_type, Instance)
-                  and (actual_type.type.fullname == 'builtins.dict')):
-                # Dict **arg.
-                # TODO: Handle arbitrary Mapping
+            elif (
+                isinstance(actual_type, Instance) and
+                len(actual_type.args) > 1 and
+                is_subtype(actual_type, self.context.mapping_type)
+            ):
+                # Only `Mapping` type can be unpacked with `**`.
+                # Other types will produce an error somewhere else.
                 return actual_type.args[1]
```

Comment:
Wouldn't this introduce a new bug if the generic args aren't the same as for Mapping? I guess that's #11138.

Reply:
Yes, I am going to solve this as a part of #11138 🙂